### ISSUE TITLE
Local signals

### ISSUE THREAD

**Issue by Dhaimadoshi** [ssannandeji/Zenject-2019#470]
Hey there, 

I am a little bit confuse at the implementation of signals.

I am trying to use the facade pattern. Each of my facades has an installer and a game object context.

So basically, I have some behaviour that can disable the logic of a facade entity. I want the facade to then send a signal of type "EnabledSignal" so that all the other components that need to be disabled can disable themselves.

My problem is that, when I have more than one instance of the facade, all the of them are notified when the signal is fired.

I am a little puzzled as for why the declaration of the Signal goes up the container chain. Can someone explain to me why ? And also, is it a bad practice to try to achieve this kind of behaviour using signals ?

I thought that I could achieve this using signal IDs, but since I need to inject the signal together with that ID using [Inject(Id = myID)], myID needs to be a static const, and so I can't get a reference to the instance...

So in short, what I want to do is having a signal with a scope bound to a subcontainer. I am thinking about implementing my own class to do that but I'm surprised it is not already present in zenject, so I was wondering if it wasn't a bad idea, please advice! 

Thank you :)

**Comment by svermeulen** [comment]
Yes this is a major flaw in the design of signals so apologies for that.  Signals work completely different in Zenject 6.  It uses an event bus pattern there (documentation coming shortly).  So unfortunately I don't think we are going to fix this problem in favour of just moving on to the new implementation.

**Comment by svermeulen** [comment]
This is a duplicate of #256 also.  But this is fixed in the new implementation.  The same signal will only be used in the container where it is declared or the children

**Comment by Dhaimadoshi** [comment]
I see, so my understanding of this was correct, thank you for the anwser !
